en_US/data-integration/snowflake.md

```diff
@@ -0,0 +1,316 @@
+
+```
+scripts/install-snowflake-driver.sh
+```
+
+#### macOS
+
+To install and configure the Snowflake ODBC driver on macOS, follow these steps:
+
+1. Install unixODBC, for example:
+
+   ```
+   brew install unixodbc
+   ```
+
+2. [Download and install iODBC](https://github.com/openlink/iODBC/releases/download/v3.52.16/iODBC-SDK-3.52.16-macOS11.dmg).
+
+3. [Download and install the Snowflake ODBC driver](https://sfc-repo.snowflakecomputing.com/odbc/macuniversal/3.3.2/snowflake_odbc_mac_64universal-3.3.2.dmg).
+
+4. Refer to [Installing and configuring the ODBC Driver for macOS](https://docs.snowflake.com/en/developer-guide/odbc/odbc-mac) for detailed installation and configuration instructions.
+
+5. After installation, update the following configuration files:
+
+   - Update permissions and configuration for the Snowflake ODBC driver:
+
+     ```bash
+     chown $(id -u):$(id -g) /opt/snowflake/snowflakeodbc/lib/universal/simba.snowflake.ini
+     echo 'ODBCInstLib=libiodbcinst.dylib' >> /opt/snowflake/snowflakeodbc/lib/universal/simba.snowflake.ini
+     ```
+
+   - Create or update the `~/.odbc.ini` file to configure the ODBC connection:
+
+     ```
+     cat << EOF > ~/.odbc.ini
+     [ODBC]
+     Trace=no
+     TraceFile=
+     
+     [ODBC Drivers]
+     Snowflake = Installed
+     
+     [ODBC Data Sources]
+     snowflake = Snowflake
+     
+     [Snowflake]
+     Driver = /opt/snowflake/snowflakeodbc/lib/universal/libSnowflake.dylib
+     EOF
+     ```
+
+### Create a User Account and Database
+
+Once the Snowflake ODBC driver is installed, you need to set up a user account, database, and related resources for data ingestion. The following credentials will be required later for configuring the connector and Sink in EMQX:
+
+| Field                  | Value                                            |
+| ---------------------- | ------------------------------------------------ |
+| Data Source Name (DSN) | `snowflake`                                      |
+| Username               | `snowpipeuser`                                   |
+| Password               | `Snowpipeuser99`                                 |
+| Database Name          | `testdatabase`                                   |
+| Schema                 | `public`                                         |
+| Stage                  | `emqx`                                           |
+| Pipe                   | `emqx`                                           |
+| Pipe User              | `snowpipeuser`                                   |
+| Private Key            | `file://<path to snowflake_rsa_key.private.pem>` |
+
+#### Generate RSA Key Pair
+
+To securely connect to Snowflake, generate an RSA key pair for authentication using the following commands:
+
+```bash
+openssl genrsa 2048 | openssl pkcs8 -topk8 -inform PEM -out snowflake_rsa_key.private.pem -nocrypt
+openssl rsa -in snowflake_rsa_key.private.pem -pubout -out snowflake_rsa_key.public.pem
+```
+
```

Comment:
I suggest adding a reference to Snowflake's docs on this matter: https://docs.snowflake.com/en/user-guide/key-pair-auth